client/lib/user/user.d.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+import { Language } from '../../languages';
+import { GravatarOptions } from '../../gravatar/types';
+import { URL, JSONSerializable } from '../../types';
+
+type WPCOMError = { message: string };
+type WPCOMHeaders = { [ name: string ]: string } & { status: number };
+
+type WPCOMCallback = (
+	error: WPCOMError | null,
+	result: JSONSerializable | null,
+	headers: WPCOMHeaders
+) => void;
+
+export interface User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: ( userId: number ) => void;
+	get: () => UserData;
+	fetch: () => Promise< void >;
+	handleFetchFailure: ( error: Error ) => void;
+	handleFetchSuccess: ( userdata: UserData ) => void;
+	getLanguage: () => Language | undefined;
+	getAvatarUrl: ( options: GravatarOptions ) => URL;
+	clear: () => Promise< void > | void;
+	sendVerificationEmail< F extends WPCOMCallback >( f: F ): XMLHttpRequest;
```

Comment:
probably best to use `callback` as the name here even though I know I used `f` in the examples. it will probably confuse people less

```ts
sendVerificationEmail<T extends WPCOMCallback>(callback: T): XMLHttpRequest;
```